crates/sequencer/src/sequencer.rs

```diff
@@ -918,14 +922,50 @@ where
     }
 
     /// Updates the given RpcModule with Sequencer methods.
-    pub async fn register_rpc_methods(
+    pub(crate) async fn register_rpc_methods(
         &self,
-        mut rpc_methods: jsonrpsee::RpcModule<()>,
-    ) -> Result<jsonrpsee::RpcModule<()>, jsonrpsee::core::RegisterMethodError> {
+        rpc_methods: jsonrpsee::RpcModule<()>,
+    ) -> Result<jsonrpsee::RpcModule<RpcContext<C>>, jsonrpsee::core::RegisterMethodError> {
         let rpc_context = self.create_rpc_context().await;
-        let rpc = create_rpc_module(rpc_context)?;
-        rpc_methods.merge(rpc)?;
-        Ok(rpc_methods)
+        let mut rpc = create_rpc_module(rpc_context)?;
+
+        // TODO: create register_healthcheck_rpc helper function
+        let ledger_db = self.ledger_db.clone();
+        rpc.register_method("health_check", move |_, _| {
+            let next_soft_batch_num = ledger_db.get_next_items_numbers().soft_batch_number;
```

Comment:
using this method might be problematic. can't you use `ledger.getHeadSoftBatch()`?